File: src/equiv/AddDigits2.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "AddDigits2"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "addDigits1"
            "addDigits3"
            [x]
            [E.mk_igt x (E.Int 0)]

File: benchmark/equiv/AddDigits2.java
public class AddDigits2 {
  static public int addDigits1(int num) {
    int result = num - 9 * ((num - 1) / 9);
    return result;
  }

  static public int addDigits3(int num) {

    while (num > 9)
      num = num / 10 + num % 10;

    return num;
  }
}
